std/format.d

```diff
@@ -583,6 +583,7 @@ can match the expected number of readings or fewer, even zero, if a
 matching failure happens.
  */
 uint formattedRead(R, Char, S...)(ref R r, const(Char)[] fmt, S args)
+    if (!S.length || is(typeof(*args[0])))
```

Comment:
I wasn't sure if only pointers are to be allowed, because one can also create custom class with unary `*` overloaded (see my unit test). I think allowing only pointers would make more sense and could potentially solve another [issue](https://issues.dlang.org/show_bug.cgi?id=8471).